scripts/lib/CIME/XML/generic_xml.py

```diff
@@ -329,21 +330,30 @@ def get_version(self):
         return version
 
     def check_timestamp(self):
+        """
+        Returns True if timestamp matches what is expected
+        """
         timestamp_cache = self._FILEMAP[self.filename].modtime
         if timestamp_cache != 0.0:
             timestamp_file  = os.path.getmtime(self.filename)
-            expect(timestamp_file == timestamp_cache,
-                   "File {} appears to have changed without a corresponding invalidation, modtimes {:0.2f} != {:0.2f}".format(self.filename, timestamp_cache, timestamp_file))
+            return timestamp_file == timestamp_cache
+        else:
+            return True
+
+    def validate_timestamp(self):
+        timestamp_ok = self.check_timestamp()
+        expect(timestamp_ok,
+               "File {} appears to have changed without a corresponding invalidation.".format(self.filename))
 
     def write(self, outfile=None, force_write=False):
         """
         Write an xml file from data in self
         """
-        #self.check_timestamp()
-
         if not (self.needsrewrite or force_write):
             return
 
+        self.validate_timestamp()
```

Comment:
Restores timestamp checking when a file gets written. I think we need this since any user changes would be blown-away by this write.